### PR TITLE
[ new ] added distance operator to Data.Nat

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
Added a new operator for distance between two natural numbers (i.e. absolute value of non-saturating subtraction). Something I've been playing about with and thought might be useful.

Comments on notation welcome.

### CODE REVIEW

**Comment by JacquesCarette** [comment]
I like the notation. Caveat: there's a chance it might not play well with some other similar notation in other modules.

**Comment by MatthewDaggitt** [comment]
> Caveat: there's a chance it might not play well with some other similar notation in other modules.

So it's true, if you import it at the same time as `Data.Integer` then the notation will clash. But that's true of almost every operator in `Data.Nat`. There weren't any resulting clashes when adding it as we always qualify the operators with the module.

**Comment by gallais** [comment]
I realised we could prove `_*_ DistributesOver ∣_-_∣` and then had to add a few
additional lemmas to actually prove it. Not sure about the name `∣n-m∣-case`.

File: src/Data/Nat/Properties.agda [hunk 1]
```diff
@@ -1129,6 +1129,93 @@ im≡jm+n⇒[i∸j]m≡n i j m n eq = begin
...
+∣n-m∣≤n⊔m (suc n) zero    = ≤-refl
+∣n-m∣≤n⊔m (suc n) (suc m) = ≤-step (∣n-m∣≤n⊔m n m)
+
+∣-∣-comm : Commutative ∣_-_∣
+∣-∣-comm zero    zero    = refl
+∣-∣-comm zero    (suc m) = refl
+∣-∣-comm (suc n) zero    = refl
+∣-∣-comm (suc n) (suc m) = ∣-∣-comm n m
+
+∣n-m∣-case : ∀ m n → (∣ n - m ∣ ≡ n ∸ m) ⊎ (∣ n - m ∣ ≡ m ∸ n)
```
[the remark this answers is not in the document]
Comment (gallais) [reply src/Data/Nat/Properties.agda]: I like this better than `-case` but the use `⊎` is maybe a bit confusing?

[the remark this answers is not in the document]
Comment (MatthewDaggitt) [reply src/Data/Nat/Properties.agda]: Yes, you're right. We usually use `∨` instead. See `i*j≡0⇒i≡0∨j≡0`. Would that be better?

**Pull request merged**